tiledb/sm/misc/test/unit_mgc_dict.cc

```diff
@@ -56,42 +55,25 @@
 
 using tiledb::sm::magic_dict;
 
-int check_embedded_data_validity() {
+TEST_CASE("Test embedded data validity", "[mgc_dict][embedded_validity]") {
   FILE* infile = nullptr;
   infile = fopen(TILEDB_PATH_TO_MAGIC_MGC, "rb");
-  if (!infile) {
-    fprintf(stderr, "ERROR: Unable to open %s\n", TILEDB_PATH_TO_MAGIC_MGC);
-    return 1;
-  }
+  REQUIRE(infile);
 
   fseek(infile, 0L, SEEK_END);
   uint64_t magic_mgc_len = ftell(infile);
   fseek(infile, 0L, SEEK_SET);
 
   char* magic_mgc_data = tdb_new_array(char, magic_mgc_len);
```

Comment:
There is never a need to allocate a local variable when the size is known in advance. Since the lengths are all `constexpr`, we can use `std::array` here.

Reply:
The uncompressed file is 6.7MB big. Won't an `std::array` here overflow the stack?

Reply:
> the lengths are all `constexpr`

I would prefer to keep the generated header an implementation detail and include it only in `magic_mgc.cc`.

Reply:
I will make it a vector; at present it leaks memory.